### DISCUSSION TITLE
timeout option for lsp request?

### DISCUSSION THREAD

**Discussion by onriv** [redhat-developer/lsp4ij#592] [General]
Hi anyway to setting timeout for a lsp request? Sometimes the textDocument/complete takes seconds in our language server

**Comment by angelozerr** [comment]
LSP4IJ doesn't manage timeout, why you need timeout? The main idea is to wait for an LSP response and cancel the LSP request when IJ cancel the feature like completion.

Do you see some performance problem?

**Reply by onriv** [reply]
> LSP4IJ doesn't manage timeout, why you need timeout? The main idea is to wait for an LSP response and cancel the LSP request when IJ cancel the feature like completion.
> 
> Do you see some performance problem?

Yeah sometime it's slow:
![image](https://github.com/user-attachments/assets/80aefab2-c75c-4077-b149-22d3a0ecef11)

It seems kind of special for the completion action. The completion candidate list can have multiple contributors, but it must wait until all contributors are completed before displaying it. I am checking if it's possible that if it's too long than I can supply a simple complete contributor. Maybe there can be some optimization from the server end though.

**Reply by angelozerr** [reply]
When you say "it is slow" you mean that completion takes time, but is IJ is freezing?

Have you the same problem with vscode?

**Reply by onriv** [reply]
The IJ is not freezing, but no completion only shown up after waiting the response received. In fact I think this is the expected behavior as the code implementation. But I kind of want to see if I can ignore the lsp complete contributor and shown only a manual contributor I defined:

```
<completion.contributor language="lean4" implementationClass="lean4ij.language.Lean4ManualCompletionContributor"
                        order="first, before commitCompletion"/>
```

Yeah vscode seems having the same problem currently. The complete candicate is quite lengthy. maybe the language server should take some optimization first.

![image](https://github.com/user-attachments/assets/f526510d-d811-48c8-a141-ea5570e6a8b3)

**Reply by onriv** [reply]
indeed this is not suitable in lsp4ij, I close it, thanks!
